routing/edge_estimator.cpp

```diff
@@ -101,6 +102,14 @@ double CarEdgeEstimator::CalcHeuristic(m2::PointD const & from, m2::PointD const
 {
   return TimeBetweenSec(from, to, m_maxSpeedMPS);
 }
+
+double CarEdgeEstimator::GetUTurnPenalty() const
+{
+  // Adds 2 minutes penalty for U-turn. The value is quite arbitrary
+  // and needs to be properly selected after a number of real-world
+  // experiments.
+  return 2 * 60;  // seconds
```

Comment:
Ok. I see. I guess it's better to think about it more before implementing it. As for U-Turn on big roads I'm talking about something like that (such lane): 

https://yandex.ru/maps/10725/domodedovo/?ll=37.752261%2C55.455879&z=15&l=stv%2Csta&panorama%5Bpoint%5D=37.758403%2C55.459277&panorama%5Bdirection%5D=340.266437%2C26.667236&panorama%5Bspan%5D=120.000000%2C77.783070